



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.4.0">
    
    
      
        <title>File Formats - Bioinformatics Tutorials</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.0284f74d.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.01803549.css">
      
      
        
        
        <meta name="theme-color" content="#e91e63">
      
    
    
      <script src="../assets/javascripts/modernizr.74668098.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
    
    
      
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="pink" data-md-color-accent="pink">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#file-formats" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href=".." title="Bioinformatics Tutorials" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Bioinformatics Tutorials
            </span>
            <span class="md-header-nav__topic">
              
                File Formats
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/HadrienG/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    HadrienG/tutorials
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href=".." title="Bioinformatics Tutorials" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Bioinformatics Tutorials
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/HadrienG/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    HadrienG/tutorials
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Individual lessons
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Individual lessons
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../command_line/" title="The command-line" class="md-nav__link">
      The command-line
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        File Formats
      </label>
    
    <a href="./" title="File Formats" class="md-nav__link md-nav__link--active">
      File Formats
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#table-of-contents" title="Table of Contents" class="md-nav__link">
    Table of Contents
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#the-fasta-format" title="The fasta format" class="md-nav__link">
    The fasta format
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-fastq-format" title="The fastq format" class="md-nav__link">
    The fastq format
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quality" title="Quality" class="md-nav__link">
    Quality
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-sambam-format" title="the sam/bam format" class="md-nav__link">
    the sam/bam format
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#example-header-section" title="Example header section" class="md-nav__link">
    Example header section
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#example-read" title="Example read" class="md-nav__link">
    Example read
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-vcf-format" title="the vcf format" class="md-nav__link">
    the vcf format
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#vcf-example" title="VCF Example" class="md-nav__link">
    VCF Example
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-gff-format" title="the gff format" class="md-nav__link">
    the gff format
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#example-gff" title="Example gff" class="md-nav__link">
    Example gff
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../qc/" title="Quality Control and Trimming" class="md-nav__link">
      Quality Control and Trimming
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../mapping/" title="Mapping and Variant Calling" class="md-nav__link">
      Mapping and Variant Calling
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../assembly/" title="De-novo Genome Assembly" class="md-nav__link">
      De-novo Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../annotation/" title="Genome Annotation" class="md-nav__link">
      Genome Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pan_genome/" title="Pan-Genome Analysis" class="md-nav__link">
      Pan-Genome Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../16S/" title="Metabarcoding" class="md-nav__link">
      Metabarcoding
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../wms/" title="Whole Metagenome Sequencing" class="md-nav__link">
      Whole Metagenome Sequencing
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../meta_assembly/" title="Metagenome assembly" class="md-nav__link">
      Metagenome assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../meta_assembly_2/" title="Metagenome assembly (continued)" class="md-nav__link">
      Metagenome assembly (continued)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../metavir/" title="Viral metagenomics" class="md-nav__link">
      Viral metagenomics
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../rna/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../nanopore/" title="Introduction to Nanopore Sequencing" class="md-nav__link">
      Introduction to Nanopore Sequencing
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#table-of-contents" title="Table of Contents" class="md-nav__link">
    Table of Contents
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#the-fasta-format" title="The fasta format" class="md-nav__link">
    The fasta format
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-fastq-format" title="The fastq format" class="md-nav__link">
    The fastq format
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quality" title="Quality" class="md-nav__link">
    Quality
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-sambam-format" title="the sam/bam format" class="md-nav__link">
    the sam/bam format
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#example-header-section" title="Example header section" class="md-nav__link">
    Example header section
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#example-read" title="Example read" class="md-nav__link">
    Example read
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-vcf-format" title="the vcf format" class="md-nav__link">
    the vcf format
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#vcf-example" title="VCF Example" class="md-nav__link">
    VCF Example
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-gff-format" title="the gff format" class="md-nav__link">
    the gff format
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#example-gff" title="Example gff" class="md-nav__link">
    Example gff
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/HadrienG/tutorials/edit/master/docs/file_formats.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="file-formats">File Formats<a class="headerlink" href="#file-formats" title="Permanent link">&para;</a></h1>
<p>This lecture is aimed at making you discover the most popular file formats used in bioinformatics. You're expected to have basic working knowledge of Linux to be able to follow the lesson.</p>
<h2 id="table-of-contents">Table of Contents<a class="headerlink" href="#table-of-contents" title="Permanent link">&para;</a></h2>
<ul>
<li><a href="#the-fasta-format">The fasta format</a></li>
<li><a href="#the-fastq-format">The fastq format</a></li>
<li><a href="#the-sam-format">The sam/bam format</a></li>
<li><a href="#the-vcf-format">The vcf format</a></li>
<li><a href="#the-gff-format">The gff format</a></li>
</ul>
<h3 id="the-fasta-format">The fasta format<a class="headerlink" href="#the-fasta-format" title="Permanent link">&para;</a></h3>
<p>The fasta format was invented in 1988 and designed to represent nucleotide or peptide sequences. It originates from the <a href="https://en.wikipedia.org/wiki/FASTA">FASTA</a> software package, but is now a standard in the world of bioinformatics.</p>
<p>The first line in a FASTA file starts with a "&gt;" (greater-than) symbol followed by the description or identifier of the sequence. Following the initial line (used for a unique description of the sequence) is the actual sequence itself in standard one-letter code.</p>
<p>A few sample sequences:</p>
<div class="codehilite"><pre><span></span><span class="o">&gt;</span><span class="n">KX580312</span><span class="p">.</span><span class="mi">1</span> <span class="n">Homo</span> <span class="n">sapiens</span> <span class="n">truncated</span> <span class="n">breast</span> <span class="n">cancer</span> <span class="mi">1</span> <span class="p">(</span><span class="n">BRCA1</span><span class="p">)</span> <span class="n">gene</span><span class="p">,</span> <span class="n">exon</span> <span class="mi">15</span> <span class="k">and</span> <span class="k">partial</span> <span class="n">cds</span>
<span class="n">GTCATCCCCTTCTAAATGCCCATCATTAGATGATAGGTGGTACATGCACAGTTGCTCTGGGAGTCTTCAG</span>
<span class="n">AATAGAAACTACCCATCTCAAGAGGAGCTCATTAAGGTTGTTGATGTGGAGGAGTAACAGCTGGAAGAGT</span>
<span class="n">CTGGGCCACACGATTTGACGGAAACATCTTACTTGCCAAGGCAAGATCTAG</span>
</pre></div>


<div class="codehilite"><pre><span></span><span class="o">&gt;</span><span class="n">KRN06561</span><span class="p">.</span><span class="mi">1</span> <span class="n">heat</span> <span class="n">shock</span> <span class="p">[</span><span class="n">Lactobacillus</span> <span class="n">sucicola</span> <span class="n">DSM</span> <span class="mi">21376</span> <span class="o">=</span> <span class="n">JCM</span> <span class="mi">15457</span><span class="p">]</span>
<span class="n">MSLVMANELTNRFNNWMKQDDFFGNLGRSFFDLDNSVNRALKTDVKETDKAYEVRIDVPGIDKKDITVDY</span>
<span class="n">HDGVLSVNAKRDSFNDESDSEGNVIASERSYGRFARQYSLPNVDESGIKAKCEDGVLKLTLPKLAEEKIN</span>
<span class="n">GNHIEIE</span>
</pre></div>


<p>A fasta file can contain multiple sequence. Each sequence will be separated by their "header" line, starting by "&gt;".</p>
<p>Example:</p>
<div class="codehilite"><pre><span></span><span class="o">&gt;</span><span class="n">KRN06561</span><span class="p">.</span><span class="mi">1</span> <span class="n">heat</span> <span class="n">shock</span> <span class="p">[</span><span class="n">Lactobacillus</span> <span class="n">sucicola</span> <span class="n">DSM</span> <span class="mi">21376</span> <span class="o">=</span> <span class="n">JCM</span> <span class="mi">15457</span><span class="p">]</span>
<span class="n">MSLVMANELTNRFNNWMKQDDFFGNLGRSFFDLDNSVNRALKTDVKETDKAYEVRIDVPGIDKKDITVDY</span>
<span class="n">HDGVLSVNAKRDSFNDESDSEGNVIASERSYGRFARQYSLPNVDESGIKAKCEDGVLKLTLPKLAEEKIN</span>
<span class="n">GNHIEIE</span>
<span class="o">&gt;</span><span class="mi">3</span><span class="n">HHU_A</span> <span class="k">Chain</span> <span class="n">A</span><span class="p">,</span> <span class="n">Human</span> <span class="n">Heat</span><span class="o">-</span><span class="n">Shock</span> <span class="n">Protein</span> <span class="mi">90</span> <span class="p">(</span><span class="n">Hsp90</span><span class="p">)</span>
<span class="n">MPEETQTQDQPMEEEEVETFAFQAEIAQLMSLIINTFYSNKEIFLRELISNSSDALDKIRYESLTDPSKL</span>
<span class="n">DSGKELHINLIPNKQDRTLTIVDTGIGMTKADLINNLGTIAKSGTKAFMEALQAGADISMIGQFGVGFYS</span>
<span class="n">AYLVAEKVTVITKHNDDEQYAWESSAGGSFTVRTDTGEPMGRGTKVILHLKEDQTEYLEERRIKEIVKKH</span>
<span class="n">SQFIGYPITLFVEK</span>
</pre></div>


<h3 id="the-fastq-format">The fastq format<a class="headerlink" href="#the-fastq-format" title="Permanent link">&para;</a></h3>
<p>The fastq format is also a text based format to represent nucleotide sequences, but also contains the corresponding quality of each nucleotide. It is the standard for storing the output of high-throughput sequencing instruments such as the Illumina machines.</p>
<p>A fastq file uses four lines per sequence:</p>
<ul>
<li>Line 1 begins with a '@' character and is followed by a sequence identifier and an optional description (like a FASTA title line).</li>
<li>Line 2 is the raw sequence letters.</li>
<li>Line 3 begins with a '+' character and is optionally followed by the same sequence identifier (and any description) again.</li>
<li>Line 4 encodes the quality values for the sequence in Line 2, and must contain the same number of symbols as letters in the sequence.</li>
</ul>
<p>An example sequence in fastq format:</p>
<div class="codehilite"><pre><span></span><span class="nv">@SEQ_ID</span><span class="w"></span>
<span class="n">GATTTGGGGTTCAAAGCAGTATCGATCAAATAGTAAATCCATTTGTTCAACTCACAGTTT</span><span class="w"></span>
<span class="o">+</span><span class="w"></span>
<span class="err">!</span><span class="s1">&#39;&#39;</span><span class="o">*</span><span class="p">((((</span><span class="o">***+</span><span class="p">))</span><span class="o">%%%++</span><span class="p">)(</span><span class="o">%%%%</span><span class="p">)</span><span class="mf">.1</span><span class="o">***-+*</span><span class="s1">&#39;&#39;</span><span class="p">))</span><span class="o">**</span><span class="mi">55</span><span class="n">CCF</span><span class="o">&gt;&gt;&gt;&gt;&gt;&gt;</span><span class="n">CCCCCCC65</span><span class="w"></span>
</pre></div>


<h4 id="quality">Quality<a class="headerlink" href="#quality" title="Permanent link">&para;</a></h4>
<p>The quality, also called phred score, is the probability that the corresponding basecall is incorrect.</p>
<p>Phred scores use a logarithmic scale, and are represented by ASCII characters, mapping to a quality usually going from 0 to 40.</p>
<table>
<thead>
<tr>
<th>Phred Quality Score</th>
<th>Probability of incorrect base call</th>
<th>Base call accuracy</th>
</tr>
</thead>
<tbody>
<tr>
<td>10</td>
<td>1 in 10</td>
<td>90%</td>
</tr>
<tr>
<td>20</td>
<td>1 in 100</td>
<td>99%</td>
</tr>
<tr>
<td>30</td>
<td>1 in 1000</td>
<td>99.9%</td>
</tr>
<tr>
<td>40</td>
<td>1 in 10,000</td>
<td>99.99%</td>
</tr>
<tr>
<td>50</td>
<td>1 in 100,000</td>
<td>99.999%</td>
</tr>
<tr>
<td>60</td>
<td>1 in 1,000,000</td>
<td>99.9999%</td>
</tr>
</tbody>
</table>
<h3 id="the-sambam-format">the sam/bam format<a class="headerlink" href="#the-sambam-format" title="Permanent link">&para;</a></h3>
<p>From <a href="https://en.wikipedia.org/wiki/SAM_(file_format)">Wikipedia</a>:</p>
<p>SAM (file format) is a text-based format for storing biological sequences aligned to a reference sequence developed by Heng Li. The acronym SAM stands for Sequence Alignment/Map. It is widely used for storing data, such as nucleotide sequences, generated by Next generation sequencing technologies and usually mapped to a reference.</p>
<p>The SAM format consists of a header and an alignment section. The binary representation of a SAM file is a BAM file, which is a compressed SAM file.[1] SAM files can be analysed and edited with the software SAMtools.</p>
<p>The SAM format has a really extensive and complex specification that you can find <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">here</a>.</p>
<p>In brief it consists of a header section and reads (with other information) in tab delimited format.</p>
<h4 id="example-header-section">Example header section<a class="headerlink" href="#example-header-section" title="Permanent link">&para;</a></h4>
<div class="codehilite"><pre><span></span><span class="nv">@HD</span><span class="w"> </span><span class="nl">VN</span><span class="p">:</span><span class="mf">1.0</span><span class="w">                  </span><span class="nl">SO</span><span class="p">:</span><span class="n">unsorted</span><span class="w"></span>
<span class="nv">@SQ</span><span class="w"> </span><span class="nl">SN</span><span class="p">:</span><span class="n">O_volvulusOVOC_OM1a</span><span class="w">  </span><span class="k">LN</span><span class="err">:</span><span class="mi">2816604</span><span class="w"></span>
<span class="nv">@SQ</span><span class="w"> </span><span class="nl">SN</span><span class="p">:</span><span class="n">O_volvulusOVOC_OM1b</span><span class="w">  </span><span class="k">LN</span><span class="err">:</span><span class="mi">28345163</span><span class="w"></span>
<span class="nv">@SQ</span><span class="w"> </span><span class="nl">SN</span><span class="p">:</span><span class="n">O_volvulusOVOC_OM2</span><span class="w">   </span><span class="k">LN</span><span class="err">:</span><span class="mi">25485961</span><span class="w"></span>
</pre></div>


<h4 id="example-read">Example read<a class="headerlink" href="#example-read" title="Permanent link">&para;</a></h4>
<div class="codehilite"><pre><span></span><span class="n">M01137</span><span class="o">:</span><span class="mi">130</span><span class="o">:</span><span class="mi">00</span><span class="o">-</span><span class="n">A</span><span class="o">:</span><span class="mi">17009</span><span class="o">:</span><span class="mi">1352</span><span class="o">/</span><span class="mi">14</span> <span class="o">*</span> <span class="mi">0</span> <span class="mi">0</span> <span class="o">*</span> <span class="o">*</span> <span class="mi">0</span> <span class="mi">0</span> <span class="n">AGCAAAATACAACGATCTGGATGGTAGCATTAGCGATGCGACACTGCTTGAACCGTCAAAG</span> <span class="n">FGGFGCFGFFGC8</span><span class="o">,,</span><span class="err">@</span><span class="n">D</span><span class="o">?</span><span class="n">E6EFCF</span><span class="o">,=</span><span class="n">AEFFGGDGGGADFGG</span><span class="err">@</span><span class="o">&gt;</span><span class="n">FFEGGG</span><span class="o">:+&lt;</span><span class="mi">7</span><span class="n">D</span><span class="o">&gt;</span><span class="n">AFCFGG</span> <span class="n">YT</span><span class="o">:</span><span class="n">Z</span><span class="o">:</span><span class="n">UU</span>
</pre></div>


<h3 id="the-vcf-format">the vcf format<a class="headerlink" href="#the-vcf-format" title="Permanent link">&para;</a></h3>
<p>The vcf format is also a text-based file format. VCF stands for Variant Call Format and is used to store gene sequence variations (SNVs, indels). The format has been developped for genotyping projects, and is the standard to represent variations in the genome of a species.</p>
<p>A vcf is a tab-delimited file, described <a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf">here</a>.</p>
<h4 id="vcf-example">VCF Example<a class="headerlink" href="#vcf-example" title="Permanent link">&para;</a></h4>
<div class="codehilite"><pre><span></span><span class="o">##</span><span class="n">fileformat</span><span class="o">=</span><span class="n">VCFv4</span><span class="p">.</span><span class="mi">0</span>
<span class="o">##</span><span class="n">fileDate</span><span class="o">=</span><span class="mi">20110705</span>
<span class="o">##</span><span class="n">reference</span><span class="o">=</span><span class="mi">1000</span><span class="n">GenomesPilot</span><span class="o">-</span><span class="n">NCBI37</span>
<span class="o">##</span><span class="n">phasing</span><span class="o">=</span><span class="k">partial</span>
<span class="o">##</span><span class="n">INFO</span><span class="o">=&lt;</span><span class="n">ID</span><span class="o">=</span><span class="n">NS</span><span class="p">,</span><span class="nb">Number</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span><span class="k">Type</span><span class="o">=</span><span class="nb">Integer</span><span class="p">,</span><span class="n">Description</span><span class="o">=</span><span class="ss">&quot;Number of Samples With Data&quot;</span><span class="o">&gt;</span>
<span class="o">##</span><span class="n">INFO</span><span class="o">=&lt;</span><span class="n">ID</span><span class="o">=</span><span class="n">DP</span><span class="p">,</span><span class="nb">Number</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span><span class="k">Type</span><span class="o">=</span><span class="nb">Integer</span><span class="p">,</span><span class="n">Description</span><span class="o">=</span><span class="ss">&quot;Total Depth&quot;</span><span class="o">&gt;</span>
<span class="o">##</span><span class="n">INFO</span><span class="o">=&lt;</span><span class="n">ID</span><span class="o">=</span><span class="n">AF</span><span class="p">,</span><span class="nb">Number</span><span class="o">=</span><span class="p">.,</span><span class="k">Type</span><span class="o">=</span><span class="nb">Float</span><span class="p">,</span><span class="n">Description</span><span class="o">=</span><span class="ss">&quot;Allele Frequency&quot;</span><span class="o">&gt;</span>
<span class="o">##</span><span class="n">INFO</span><span class="o">=&lt;</span><span class="n">ID</span><span class="o">=</span><span class="n">AA</span><span class="p">,</span><span class="nb">Number</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span><span class="k">Type</span><span class="o">=</span><span class="n">String</span><span class="p">,</span><span class="n">Description</span><span class="o">=</span><span class="ss">&quot;Ancestral Allele&quot;</span><span class="o">&gt;</span>
<span class="o">##</span><span class="n">INFO</span><span class="o">=&lt;</span><span class="n">ID</span><span class="o">=</span><span class="n">DB</span><span class="p">,</span><span class="nb">Number</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="k">Type</span><span class="o">=</span><span class="n">Flag</span><span class="p">,</span><span class="n">Description</span><span class="o">=</span><span class="ss">&quot;dbSNP membership, build 129&quot;</span><span class="o">&gt;</span>
<span class="o">##</span><span class="n">INFO</span><span class="o">=&lt;</span><span class="n">ID</span><span class="o">=</span><span class="n">H2</span><span class="p">,</span><span class="nb">Number</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="k">Type</span><span class="o">=</span><span class="n">Flag</span><span class="p">,</span><span class="n">Description</span><span class="o">=</span><span class="ss">&quot;HapMap2 membership&quot;</span><span class="o">&gt;</span>
<span class="o">##</span><span class="n">FILTER</span><span class="o">=&lt;</span><span class="n">ID</span><span class="o">=</span><span class="n">q10</span><span class="p">,</span><span class="n">Description</span><span class="o">=</span><span class="ss">&quot;Quality below 10&quot;</span><span class="o">&gt;</span>
<span class="o">##</span><span class="n">FILTER</span><span class="o">=&lt;</span><span class="n">ID</span><span class="o">=</span><span class="n">s50</span><span class="p">,</span><span class="n">Description</span><span class="o">=</span><span class="ss">&quot;Less than 50% of samples have data&quot;</span><span class="o">&gt;</span>
<span class="o">##</span><span class="n">FORMAT</span><span class="o">=&lt;</span><span class="n">ID</span><span class="o">=</span><span class="n">GQ</span><span class="p">,</span><span class="nb">Number</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span><span class="k">Type</span><span class="o">=</span><span class="nb">Integer</span><span class="p">,</span><span class="n">Description</span><span class="o">=</span><span class="ss">&quot;Genotype Quality&quot;</span><span class="o">&gt;</span>
<span class="o">##</span><span class="n">FORMAT</span><span class="o">=&lt;</span><span class="n">ID</span><span class="o">=</span><span class="n">GT</span><span class="p">,</span><span class="nb">Number</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span><span class="k">Type</span><span class="o">=</span><span class="n">String</span><span class="p">,</span><span class="n">Description</span><span class="o">=</span><span class="ss">&quot;Genotype&quot;</span><span class="o">&gt;</span>
<span class="o">##</span><span class="n">FORMAT</span><span class="o">=&lt;</span><span class="n">ID</span><span class="o">=</span><span class="n">DP</span><span class="p">,</span><span class="nb">Number</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span><span class="k">Type</span><span class="o">=</span><span class="nb">Integer</span><span class="p">,</span><span class="n">Description</span><span class="o">=</span><span class="ss">&quot;Read Depth&quot;</span><span class="o">&gt;</span>
<span class="o">##</span><span class="n">FORMAT</span><span class="o">=&lt;</span><span class="n">ID</span><span class="o">=</span><span class="n">HQ</span><span class="p">,</span><span class="nb">Number</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span><span class="k">Type</span><span class="o">=</span><span class="nb">Integer</span><span class="p">,</span><span class="n">Description</span><span class="o">=</span><span class="ss">&quot;Haplotype Quality&quot;</span><span class="o">&gt;</span>
<span class="o">#</span><span class="n">CHROM</span> <span class="n">POS</span>    <span class="n">ID</span>        <span class="k">REF</span>  <span class="n">ALT</span>     <span class="n">QUAL</span> <span class="n">FILTER</span> <span class="n">INFO</span>                              <span class="n">FORMAT</span>      <span class="n">Sample1</span>        <span class="n">Sample2</span>        <span class="n">Sample3</span>
<span class="mi">2</span>      <span class="mi">4370</span>   <span class="n">rs6057</span>    <span class="k">G</span>    <span class="n">A</span>       <span class="mi">29</span>   <span class="p">.</span>      <span class="n">NS</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">DP</span><span class="o">=</span><span class="mi">13</span><span class="p">;</span><span class="n">AF</span><span class="o">=</span><span class="mi">0</span><span class="p">.</span><span class="mi">5</span><span class="p">;</span><span class="n">DB</span><span class="p">;</span><span class="n">H2</span>           <span class="n">GT</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">HQ</span> <span class="mi">0</span><span class="o">|</span><span class="mi">0</span><span class="p">:</span><span class="mi">48</span><span class="p">:</span><span class="mi">1</span><span class="p">:</span><span class="mi">52</span><span class="p">,</span><span class="mi">51</span> <span class="mi">1</span><span class="o">|</span><span class="mi">0</span><span class="p">:</span><span class="mi">48</span><span class="p">:</span><span class="mi">8</span><span class="p">:</span><span class="mi">51</span><span class="p">,</span><span class="mi">51</span> <span class="mi">1</span><span class="o">/</span><span class="mi">1</span><span class="p">:</span><span class="mi">43</span><span class="p">:</span><span class="mi">5</span><span class="p">:.,.</span>
<span class="mi">2</span>      <span class="mi">7330</span>   <span class="p">.</span>         <span class="n">T</span>    <span class="n">A</span>       <span class="mi">3</span>    <span class="n">q10</span>    <span class="n">NS</span><span class="o">=</span><span class="mi">5</span><span class="p">;</span><span class="n">DP</span><span class="o">=</span><span class="mi">12</span><span class="p">;</span><span class="n">AF</span><span class="o">=</span><span class="mi">0</span><span class="p">.</span><span class="mi">017</span>               <span class="n">GT</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">HQ</span> <span class="mi">0</span><span class="o">|</span><span class="mi">0</span><span class="p">:</span><span class="mi">46</span><span class="p">:</span><span class="mi">3</span><span class="p">:</span><span class="mi">58</span><span class="p">,</span><span class="mi">50</span> <span class="mi">0</span><span class="o">|</span><span class="mi">1</span><span class="p">:</span><span class="mi">3</span><span class="p">:</span><span class="mi">5</span><span class="p">:</span><span class="mi">65</span><span class="p">,</span><span class="mi">3</span>   <span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">:</span><span class="mi">41</span><span class="p">:</span><span class="mi">3</span>
<span class="mi">2</span>      <span class="mi">110696</span> <span class="n">rs6055</span>    <span class="n">A</span>    <span class="k">G</span><span class="p">,</span><span class="n">T</span>     <span class="mi">67</span>   <span class="n">PASS</span>   <span class="n">NS</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">DP</span><span class="o">=</span><span class="mi">10</span><span class="p">;</span><span class="n">AF</span><span class="o">=</span><span class="mi">0</span><span class="p">.</span><span class="mi">333</span><span class="p">,</span><span class="mi">0</span><span class="p">.</span><span class="mi">667</span><span class="p">;</span><span class="n">AA</span><span class="o">=</span><span class="n">T</span><span class="p">;</span><span class="n">DB</span> <span class="n">GT</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">HQ</span> <span class="mi">1</span><span class="o">|</span><span class="mi">2</span><span class="p">:</span><span class="mi">21</span><span class="p">:</span><span class="mi">6</span><span class="p">:</span><span class="mi">23</span><span class="p">,</span><span class="mi">27</span> <span class="mi">2</span><span class="o">|</span><span class="mi">1</span><span class="p">:</span><span class="mi">2</span><span class="p">:</span><span class="mi">0</span><span class="p">:</span><span class="mi">18</span><span class="p">,</span><span class="mi">2</span>   <span class="mi">2</span><span class="o">/</span><span class="mi">2</span><span class="p">:</span><span class="mi">35</span><span class="p">:</span><span class="mi">4</span>
<span class="mi">2</span>      <span class="mi">130237</span> <span class="p">.</span>         <span class="n">T</span>    <span class="p">.</span>       <span class="mi">47</span>   <span class="p">.</span>      <span class="n">NS</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">DP</span><span class="o">=</span><span class="mi">16</span><span class="p">;</span><span class="n">AA</span><span class="o">=</span><span class="n">T</span>                   <span class="n">GT</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">HQ</span> <span class="mi">0</span><span class="o">|</span><span class="mi">0</span><span class="p">:</span><span class="mi">54</span><span class="p">:</span><span class="mi">7</span><span class="p">:</span><span class="mi">56</span><span class="p">,</span><span class="mi">60</span> <span class="mi">0</span><span class="o">|</span><span class="mi">0</span><span class="p">:</span><span class="mi">48</span><span class="p">:</span><span class="mi">4</span><span class="p">:</span><span class="mi">56</span><span class="p">,</span><span class="mi">51</span> <span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">:</span><span class="mi">61</span><span class="p">:</span><span class="mi">2</span>
<span class="mi">2</span>      <span class="mi">134567</span> <span class="n">microsat1</span> <span class="n">GTCT</span> <span class="k">G</span><span class="p">,</span><span class="n">GTACT</span> <span class="mi">50</span>   <span class="n">PASS</span>   <span class="n">NS</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">DP</span><span class="o">=</span><span class="mi">9</span><span class="p">;</span><span class="n">AA</span><span class="o">=</span><span class="k">G</span>                    <span class="n">GT</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">DP</span>    <span class="mi">0</span><span class="o">/</span><span class="mi">1</span><span class="p">:</span><span class="mi">35</span><span class="p">:</span><span class="mi">4</span>       <span class="mi">0</span><span class="o">/</span><span class="mi">2</span><span class="p">:</span><span class="mi">17</span><span class="p">:</span><span class="mi">2</span>       <span class="mi">1</span><span class="o">/</span><span class="mi">1</span><span class="p">:</span><span class="mi">40</span><span class="p">:</span><span class="mi">3</span>
<span class="n">chr1</span>    <span class="mi">45796269</span>        <span class="p">.</span>       <span class="k">G</span>       <span class="k">C</span>
<span class="n">chr1</span>    <span class="mi">45797505</span>        <span class="p">.</span>       <span class="k">C</span>       <span class="k">G</span>
<span class="n">chr1</span>    <span class="mi">45798555</span>        <span class="p">.</span>       <span class="n">T</span>       <span class="k">C</span>
<span class="n">chr1</span>    <span class="mi">45798901</span>        <span class="p">.</span>       <span class="k">C</span>       <span class="n">T</span>
<span class="n">chr1</span>    <span class="mi">45805566</span>        <span class="p">.</span>       <span class="k">G</span>       <span class="k">C</span>
<span class="n">chr2</span>    <span class="mi">47703379</span>        <span class="p">.</span>       <span class="k">C</span>       <span class="n">T</span>
<span class="n">chr2</span>    <span class="mi">48010488</span>        <span class="p">.</span>       <span class="k">G</span>       <span class="n">A</span>
<span class="n">chr2</span>    <span class="mi">48030838</span>        <span class="p">.</span>       <span class="n">A</span>       <span class="n">T</span>
<span class="n">chr2</span>    <span class="mi">48032875</span>        <span class="p">.</span>       <span class="n">CTAT</span>    <span class="o">-</span>
<span class="n">chr2</span>    <span class="mi">48032937</span>        <span class="p">.</span>       <span class="n">T</span>       <span class="k">C</span>
<span class="n">chr2</span>    <span class="mi">48033273</span>        <span class="p">.</span>       <span class="n">TTTTTGTTTTAATTCCT</span>       <span class="o">-</span>
<span class="n">chr2</span>    <span class="mi">48033551</span>        <span class="p">.</span>       <span class="k">C</span>       <span class="k">G</span>
<span class="n">chr2</span>    <span class="mi">48033910</span>        <span class="p">.</span>       <span class="n">A</span>       <span class="n">T</span>
<span class="n">chr2</span>    <span class="mi">215632048</span>       <span class="p">.</span>       <span class="k">G</span>       <span class="n">T</span>
<span class="n">chr2</span>    <span class="mi">215632125</span>       <span class="p">.</span>       <span class="n">TT</span>      <span class="o">-</span>
<span class="n">chr2</span>    <span class="mi">215632155</span>       <span class="p">.</span>       <span class="n">T</span>       <span class="k">C</span>
<span class="n">chr2</span>    <span class="mi">215632192</span>       <span class="p">.</span>       <span class="k">G</span>       <span class="n">A</span>
<span class="n">chr2</span>    <span class="mi">215632255</span>       <span class="p">.</span>       <span class="n">CA</span>      <span class="n">TG</span>
<span class="n">chr2</span>    <span class="mi">215634055</span>       <span class="p">.</span>       <span class="k">C</span>       <span class="n">T</span>
</pre></div>


<h3 id="the-gff-format">the gff format<a class="headerlink" href="#the-gff-format" title="Permanent link">&para;</a></h3>
<p>The general feature format (gff) is another text file format, used for describing genes and other features of DNA, RNA and protein sequences. It is the standard for annotation of genomes.</p>
<p>A gff file should contain 9 columns, described <a href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md">here</a></p>
<h4 id="example-gff">Example gff<a class="headerlink" href="#example-gff" title="Permanent link">&para;</a></h4>
<div class="codehilite"><pre><span></span><span class="n">##description</span><span class="err">:</span><span class="w"> </span><span class="n">evidence</span><span class="o">-</span><span class="n">based</span><span class="w"> </span><span class="n">annotation</span><span class="w"> </span><span class="k">of</span><span class="w"> </span><span class="n">the</span><span class="w"> </span><span class="n">human</span><span class="w"> </span><span class="n">genome</span><span class="w"> </span><span class="p">(</span><span class="n">GRCh38</span><span class="p">),</span><span class="w"> </span><span class="n">version</span><span class="w"> </span><span class="mi">25</span><span class="w"> </span><span class="p">(</span><span class="n">Ensembl</span><span class="w"> </span><span class="mi">85</span><span class="p">)</span><span class="w"></span>
<span class="n">##provider</span><span class="err">:</span><span class="w"> </span><span class="n">GENCODE</span><span class="w"></span>
<span class="n">##contact</span><span class="err">:</span><span class="w"> </span><span class="n">gencode</span><span class="o">-</span><span class="n">help</span><span class="nv">@sanger</span><span class="p">.</span><span class="n">ac</span><span class="p">.</span><span class="n">uk</span><span class="w"></span>
<span class="n">##format</span><span class="err">:</span><span class="w"> </span><span class="n">gtf</span><span class="w"></span>
<span class="n">##date</span><span class="err">:</span><span class="w"> </span><span class="mi">2016</span><span class="o">-</span><span class="mi">07</span><span class="o">-</span><span class="mi">15</span><span class="w"></span>
<span class="n">chr1</span><span class="w">    </span><span class="n">HAVANA</span><span class="w">  </span><span class="n">gene</span><span class="w">    </span><span class="mi">11869</span><span class="w">   </span><span class="mi">14409</span><span class="w">   </span><span class="p">.</span><span class="w">   </span><span class="o">+</span><span class="w">   </span><span class="p">.</span><span class="w">   </span><span class="n">gene_id</span><span class="w"> </span><span class="ss">&quot;ENSG00000223972.5&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_type</span><span class="w"> </span><span class="ss">&quot;transcribed_unprocessed_pseudogene&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_status</span><span class="w"> </span><span class="ss">&quot;KNOWN&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_name</span><span class="w"> </span><span class="ss">&quot;DDX11L1&quot;</span><span class="p">;</span><span class="w"> </span><span class="k">level</span><span class="w"> </span><span class="mi">2</span><span class="p">;</span><span class="w"> </span><span class="n">havana_gene</span><span class="w"> </span><span class="ss">&quot;OTTHUMG00000000961.2&quot;</span><span class="p">;</span><span class="w"></span>
<span class="n">chr1</span><span class="w">    </span><span class="n">HAVANA</span><span class="w">  </span><span class="n">transcript</span><span class="w">  </span><span class="mi">11869</span><span class="w">   </span><span class="mi">14409</span><span class="w">   </span><span class="p">.</span><span class="w">   </span><span class="o">+</span><span class="w">   </span><span class="p">.</span><span class="w">   </span><span class="n">gene_id</span><span class="w"> </span><span class="ss">&quot;ENSG00000223972.5&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_id</span><span class="w"> </span><span class="ss">&quot;ENST00000456328.2&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_type</span><span class="w"> </span><span class="ss">&quot;transcribed_unprocessed_pseudogene&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_status</span><span class="w"> </span><span class="ss">&quot;KNOWN&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_name</span><span class="w"> </span><span class="ss">&quot;DDX11L1&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_type</span><span class="w"> </span><span class="ss">&quot;processed_transcript&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_status</span><span class="w"> </span><span class="ss">&quot;KNOWN&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_name</span><span class="w"> </span><span class="ss">&quot;DDX11L1-002&quot;</span><span class="p">;</span><span class="w"> </span><span class="k">level</span><span class="w"> </span><span class="mi">2</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_support_level</span><span class="w"> </span><span class="ss">&quot;1&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">tag</span><span class="w"> </span><span class="ss">&quot;basic&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">havana_gene</span><span class="w"> </span><span class="ss">&quot;OTTHUMG00000000961.2&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">havana_transcript</span><span class="w"> </span><span class="ss">&quot;OTTHUMT00000362751.1&quot;</span><span class="p">;</span><span class="w"></span>
<span class="n">chr1</span><span class="w">    </span><span class="n">HAVANA</span><span class="w">  </span><span class="n">exon</span><span class="w">    </span><span class="mi">11869</span><span class="w">   </span><span class="mi">12227</span><span class="w">   </span><span class="p">.</span><span class="w">   </span><span class="o">+</span><span class="w">   </span><span class="p">.</span><span class="w">   </span><span class="n">gene_id</span><span class="w"> </span><span class="ss">&quot;ENSG00000223972.5&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_id</span><span class="w"> </span><span class="ss">&quot;ENST00000456328.2&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_type</span><span class="w"> </span><span class="ss">&quot;transcribed_unprocessed_pseudogene&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_status</span><span class="w"> </span><span class="ss">&quot;KNOWN&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_name</span><span class="w"> </span><span class="ss">&quot;DDX11L1&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_type</span><span class="w"> </span><span class="ss">&quot;processed_transcript&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_status</span><span class="w"> </span><span class="ss">&quot;KNOWN&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_name</span><span class="w"> </span><span class="ss">&quot;DDX11L1-002&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">exon_number</span><span class="w"> </span><span class="mi">1</span><span class="p">;</span><span class="w"> </span><span class="n">exon_id</span><span class="w"> </span><span class="ss">&quot;ENSE00002234944.1&quot;</span><span class="p">;</span><span class="w"> </span><span class="k">level</span><span class="w"> </span><span class="mi">2</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_support_level</span><span class="w"> </span><span class="ss">&quot;1&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">tag</span><span class="w"> </span><span class="ss">&quot;basic&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">havana_gene</span><span class="w"> </span><span class="ss">&quot;OTTHUMG00000000961.2&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">havana_transcript</span><span class="w"> </span><span class="ss">&quot;OTTHUMT00000362751.1&quot;</span><span class="p">;</span><span class="w"></span>
<span class="n">chr1</span><span class="w">    </span><span class="n">HAVANA</span><span class="w">  </span><span class="n">exon</span><span class="w">    </span><span class="mi">12613</span><span class="w">   </span><span class="mi">12721</span><span class="w">   </span><span class="p">.</span><span class="w">   </span><span class="o">+</span><span class="w">   </span><span class="p">.</span><span class="w">   </span><span class="n">gene_id</span><span class="w"> </span><span class="ss">&quot;ENSG00000223972.5&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_id</span><span class="w"> </span><span class="ss">&quot;ENST00000456328.2&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_type</span><span class="w"> </span><span class="ss">&quot;transcribed_unprocessed_pseudogene&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_status</span><span class="w"> </span><span class="ss">&quot;KNOWN&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_name</span><span class="w"> </span><span class="ss">&quot;DDX11L1&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_type</span><span class="w"> </span><span class="ss">&quot;processed_transcript&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_status</span><span class="w"> </span><span class="ss">&quot;KNOWN&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_name</span><span class="w"> </span><span class="ss">&quot;DDX11L1-002&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">exon_number</span><span class="w"> </span><span class="mi">2</span><span class="p">;</span><span class="w"> </span><span class="n">exon_id</span><span class="w"> </span><span class="ss">&quot;ENSE00003582793.1&quot;</span><span class="p">;</span><span class="w"> </span><span class="k">level</span><span class="w"> </span><span class="mi">2</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_support_level</span><span class="w"> </span><span class="ss">&quot;1&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">tag</span><span class="w"> </span><span class="ss">&quot;basic&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">havana_gene</span><span class="w"> </span><span class="ss">&quot;OTTHUMG00000000961.2&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">havana_transcript</span><span class="w"> </span><span class="ss">&quot;OTTHUMT00000362751.1&quot;</span><span class="p">;</span><span class="w"></span>
<span class="n">chr1</span><span class="w">    </span><span class="n">HAVANA</span><span class="w">  </span><span class="n">exon</span><span class="w">    </span><span class="mi">13221</span><span class="w">   </span><span class="mi">14409</span><span class="w">   </span><span class="p">.</span><span class="w">   </span><span class="o">+</span><span class="w">   </span><span class="p">.</span><span class="w">   </span><span class="n">gene_id</span><span class="w"> </span><span class="ss">&quot;ENSG00000223972.5&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_id</span><span class="w"> </span><span class="ss">&quot;ENST00000456328.2&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_type</span><span class="w"> </span><span class="ss">&quot;transcribed_unprocessed_pseudogene&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_status</span><span class="w"> </span><span class="ss">&quot;KNOWN&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">gene_name</span><span class="w"> </span><span class="ss">&quot;DDX11L1&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_type</span><span class="w"> </span><span class="ss">&quot;processed_transcript&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_status</span><span class="w"> </span><span class="ss">&quot;KNOWN&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_name</span><span class="w"> </span><span class="ss">&quot;DDX11L1-002&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">exon_number</span><span class="w"> </span><span class="mi">3</span><span class="p">;</span><span class="w"> </span><span class="n">exon_id</span><span class="w"> </span><span class="ss">&quot;ENSE00002312635.1&quot;</span><span class="p">;</span><span class="w"> </span><span class="k">level</span><span class="w"> </span><span class="mi">2</span><span class="p">;</span><span class="w"> </span><span class="n">transcript_support_level</span><span class="w"> </span><span class="ss">&quot;1&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">tag</span><span class="w"> </span><span class="ss">&quot;basic&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">havana_gene</span><span class="w"> </span><span class="ss">&quot;OTTHUMG00000000961.2&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">havana_transcript</span><span class="w"> </span><span class="ss">&quot;OTTHUMT00000362751.1&quot;</span><span class="p">;</span><span class="w"></span>
</pre></div>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../command_line/" title="The command-line" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                The command-line
              </span>
            </div>
          </a>
        
        
          <a href="../qc/" title="Quality Control and Trimming" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Quality Control and Trimming
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
  <div class="md-footer-social">
    <link rel="stylesheet" href="../assets/fonts/font-awesome.css">
    
      <a href="https://github.com/HadrienG" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://twitter.com/HGourle" class="md-footer-social__link fa fa-twitter"></a>
    
  </div>

    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.245445c6.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
    
  </body>
</html>