



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.4.0">
    
    
      
        <title>Metabarcoding - Bioinformatics Tutorials@SLU BI1324 (30224), 2020</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.0284f74d.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.01803549.css">
      
      
        
        
        <meta name="theme-color" content="#2196f3">
      
    
    
      <script src="../assets/javascripts/modernizr.74668098.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
    
    
      
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="blue" data-md-color-accent="blue">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#metabarcoding" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href=".." title="Bioinformatics Tutorials@SLU BI1324 (30224), 2020" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Bioinformatics Tutorials@SLU BI1324 (30224), 2020
            </span>
            <span class="md-header-nav__topic">
              
                Metabarcoding
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/domenico-simone/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    domenico-simone/tutorials
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href=".." title="Bioinformatics Tutorials@SLU BI1324 (30224), 2020" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Bioinformatics Tutorials@SLU BI1324 (30224), 2020
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/domenico-simone/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    domenico-simone/tutorials
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Individual lessons
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Individual lessons
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../command_line/" title="The command-line" class="md-nav__link">
      The command-line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../file_formats/" title="File Formats" class="md-nav__link">
      File Formats
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../qc/" title="Quality Control and Trimming" class="md-nav__link">
      Quality Control and Trimming
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../mapping/" title="Mapping and Variant Calling" class="md-nav__link">
      Mapping and Variant Calling
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../assembly/" title="De-novo Genome Assembly" class="md-nav__link">
      De-novo Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../annotation/" title="Genome Annotation" class="md-nav__link">
      Genome Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pan_genome/" title="Pan-Genome Analysis" class="md-nav__link">
      Pan-Genome Analysis
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Metabarcoding
      </label>
    
    <a href="./" title="Metabarcoding" class="md-nav__link md-nav__link--active">
      Metabarcoding
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#before-starting" title="Before Starting" class="md-nav__link">
    Before Starting
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-and-load-packages" title="Install and Load Packages" class="md-nav__link">
    Install and Load Packages
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#download-the-data" title="Download the Data" class="md-nav__link">
    Download the Data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#filtering-and-trimming" title="Filtering and Trimming" class="md-nav__link">
    Filtering and Trimming
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learn-the-error-rates" title="Learn the Error Rates" class="md-nav__link">
    Learn the Error Rates
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dereplication" title="Dereplication" class="md-nav__link">
    Dereplication
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sample-inference" title="Sample inference" class="md-nav__link">
    Sample inference
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#merge-paired-end-reads" title="Merge Paired-end Reads" class="md-nav__link">
    Merge Paired-end Reads
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#construct-sequence-table" title="Construct Sequence Table" class="md-nav__link">
    Construct Sequence Table
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#remove-chimeras" title="Remove Chimeras" class="md-nav__link">
    Remove Chimeras
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assign-taxonomy" title="Assign Taxonomy" class="md-nav__link">
    Assign Taxonomy
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#phylogenetic-tree" title="Phylogenetic Tree" class="md-nav__link">
    Phylogenetic Tree
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#phyloseq" title="Phyloseq" class="md-nav__link">
    Phyloseq
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../wms/" title="Whole Metagenome Sequencing" class="md-nav__link">
      Whole Metagenome Sequencing
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../meta_assembly/" title="Metagenome assembly" class="md-nav__link">
      Metagenome assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../meta_assembly_2/" title="Metagenome assembly (continued)" class="md-nav__link">
      Metagenome assembly (continued)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../metavir/" title="Viral metagenomics" class="md-nav__link">
      Viral metagenomics
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../rna/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../nanopore/" title="Introduction to Nanopore Sequencing" class="md-nav__link">
      Introduction to Nanopore Sequencing
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#before-starting" title="Before Starting" class="md-nav__link">
    Before Starting
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-and-load-packages" title="Install and Load Packages" class="md-nav__link">
    Install and Load Packages
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#download-the-data" title="Download the Data" class="md-nav__link">
    Download the Data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#filtering-and-trimming" title="Filtering and Trimming" class="md-nav__link">
    Filtering and Trimming
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learn-the-error-rates" title="Learn the Error Rates" class="md-nav__link">
    Learn the Error Rates
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dereplication" title="Dereplication" class="md-nav__link">
    Dereplication
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sample-inference" title="Sample inference" class="md-nav__link">
    Sample inference
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#merge-paired-end-reads" title="Merge Paired-end Reads" class="md-nav__link">
    Merge Paired-end Reads
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#construct-sequence-table" title="Construct Sequence Table" class="md-nav__link">
    Construct Sequence Table
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#remove-chimeras" title="Remove Chimeras" class="md-nav__link">
    Remove Chimeras
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assign-taxonomy" title="Assign Taxonomy" class="md-nav__link">
    Assign Taxonomy
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#phylogenetic-tree" title="Phylogenetic Tree" class="md-nav__link">
    Phylogenetic Tree
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#phyloseq" title="Phyloseq" class="md-nav__link">
    Phyloseq
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/domenico-simone/tutorials/edit/master/docs/16S.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="metabarcoding">Metabarcoding<a class="headerlink" href="#metabarcoding" title="Permanent link">&para;</a></h1>
<p>This tutorial is aimed at being a walkthrough of the DADA2 pipeline.
It uses the data of the now famous <a href="http://www.mothur.org/wiki/MiSeq_SOP">MiSeq SOP</a> by the Mothur authors but analyses the data using DADA2.</p>
<p>DADA2 is a relatively new method to analyse amplicon data which uses exact variants instead of OTUs.    </p>
<p>The advantages of the DADA2 method is described in the <a href="http://dx.doi.org/10.1038/ismej.2017.119">paper</a></p>
<h2 id="before-starting">Before Starting<a class="headerlink" href="#before-starting" title="Permanent link">&para;</a></h2>
<p>There are two ways to follow this tutorial: you can copy and paste all the codes blocks below in R directly, or you can download this document in the Rmarkdown format and execute the cells.</p>
<ul>
<li><a href="../data/16S.Rmd">Link to the document in Rmarkdown</a></li>
</ul>
<h2 id="install-and-load-packages">Install and Load Packages<a class="headerlink" href="#install-and-load-packages" title="Permanent link">&para;</a></h2>
<p>First install DADA2 and other necessary packages</p>
<div class="codehilite"><pre><span></span><span class="nf">source</span><span class="p">(</span><span class="s">&#39;https://bioconductor.org/biocLite.R&#39;</span><span class="p">)</span>
<span class="nf">biocLite</span><span class="p">(</span><span class="s">&#39;dada2&#39;</span><span class="p">)</span>
<span class="nf">biocLite</span><span class="p">(</span><span class="s">&#39;phyloseq&#39;</span><span class="p">)</span>
<span class="nf">biocLite</span><span class="p">(</span><span class="s">&#39;DECIPHER&#39;</span><span class="p">)</span>
<span class="nf">install.packages</span><span class="p">(</span><span class="s">&#39;ggplot2&#39;</span><span class="p">)</span>
<span class="nf">install.packages</span><span class="p">(</span><span class="s">&#39;phangorn&#39;</span><span class="p">)</span>
</pre></div>


<p>Now load the packages and verify you have the correct DADA2 version</p>
<div class="codehilite"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">dada2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">phyloseq</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">phangorn</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">DECIPHER</span><span class="p">)</span>
<span class="nf">packageVersion</span><span class="p">(</span><span class="s">&#39;dada2&#39;</span><span class="p">)</span>
</pre></div>


<h2 id="download-the-data">Download the Data<a class="headerlink" href="#download-the-data" title="Permanent link">&para;</a></h2>
<p>You will also need to download the data, as well as the SILVA database</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>If you are following the tutorial on the website, the following block of commands has to be executed outside of R.
If you run this tutorial with the R notebook, you can simply execute the cell block</p>
</div>
<div class="codehilite"><pre><span></span>wget http://www.mothur.org/w/images/d/d6/MiSeqSOPData.zip
unzip MiSeqSOPData.zip
rm -r __MACOSX/
<span class="nb">cd</span> MiSeq_SOP
wget https://zenodo.org/record/824551/files/silva_nr_v128_train_set.fa.gz
wget https://zenodo.org/record/824551/files/silva_species_assignment_v128.fa.gz
<span class="nb">cd</span> ..
</pre></div>


<p>Back in R, check that you have downloaded the data</p>
<div class="codehilite"><pre><span></span><span class="n">path</span> <span class="o">&lt;-</span> <span class="s">&#39;MiSeq_SOP&#39;</span>
<span class="nf">list.files</span><span class="p">(</span><span class="n">path</span><span class="p">)</span>
</pre></div>


<h2 id="filtering-and-trimming">Filtering and Trimming<a class="headerlink" href="#filtering-and-trimming" title="Permanent link">&para;</a></h2>
<p>First we create two lists with the sorted name of the reads: one for the forward reads, one for the reverse reads</p>
<div class="codehilite"><pre><span></span><span class="n">raw_forward</span> <span class="o">&lt;-</span> <span class="nf">sort</span><span class="p">(</span><span class="nf">list.files</span><span class="p">(</span><span class="n">path</span><span class="p">,</span> <span class="n">pattern</span><span class="o">=</span><span class="s">&quot;_R1_001.fastq&quot;</span><span class="p">,</span>
                               <span class="n">full.names</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">))</span>

<span class="n">raw_reverse</span> <span class="o">&lt;-</span> <span class="nf">sort</span><span class="p">(</span><span class="nf">list.files</span><span class="p">(</span><span class="n">path</span><span class="p">,</span> <span class="n">pattern</span><span class="o">=</span><span class="s">&quot;_R2_001.fastq&quot;</span><span class="p">,</span>
                               <span class="n">full.names</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">))</span>

<span class="c1"># we also need the sample names</span>
<span class="n">sample_names</span> <span class="o">&lt;-</span> <span class="nf">sapply</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="nf">basename</span><span class="p">(</span><span class="n">raw_forward</span><span class="p">),</span> <span class="s">&quot;_&quot;</span><span class="p">),</span>
                       <span class="n">`[`</span><span class="p">,</span>  <span class="c1"># extracts the first element of a subset</span>
                       <span class="m">1</span><span class="p">)</span>
</pre></div>


<p>then we visualise the quality of our reads</p>
<div class="codehilite"><pre><span></span><span class="nf">plotQualityProfile</span><span class="p">(</span><span class="n">raw_forward[1</span><span class="o">:</span><span class="m">2</span><span class="n">]</span><span class="p">)</span>
<span class="nf">plotQualityProfile</span><span class="p">(</span><span class="n">raw_reverse[1</span><span class="o">:</span><span class="m">2</span><span class="n">]</span><span class="p">)</span>
</pre></div>


<div class="admonition question">
<p class="admonition-title">Question</p>
<p>What do you think of the read quality?</p>
</div>
<p>The forward reads are good quality (although dropping a bit at the end as usual) while the reverse are way worse.</p>
<p>Based on these profiles, we will truncate the forward reads at position 240 and the reverse reads at position 160 where the quality distribution crashes.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>in this tutorial we perform the trimming using DADA2's own functions.
If you wish to do it outside of DADA2, you can refer to the <a href="../qc/">Quality Control tutorial</a></p>
</div>
<hr />
<p>Dada2 requires us to define the name of our output files</p>
<div class="codehilite"><pre><span></span><span class="c1"># place filtered files in filtered/ subdirectory</span>
<span class="n">filtered_path</span> <span class="o">&lt;-</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">path</span><span class="p">,</span> <span class="s">&quot;filtered&quot;</span><span class="p">)</span>

<span class="n">filtered_forward</span> <span class="o">&lt;-</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">filtered_path</span><span class="p">,</span>
                              <span class="nf">paste0</span><span class="p">(</span><span class="n">sample_names</span><span class="p">,</span> <span class="s">&quot;_R1_trimmed.fastq.gz&quot;</span><span class="p">))</span>

<span class="n">filtered_reverse</span> <span class="o">&lt;-</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">filtered_path</span><span class="p">,</span>
                              <span class="nf">paste0</span><span class="p">(</span><span class="n">sample_names</span><span class="p">,</span> <span class="s">&quot;_R2_trimmed.fastq.gz&quot;</span><span class="p">))</span>
</pre></div>


<p>We’ll use standard filtering parameters: <code>maxN=0</code> (DADA22 requires no Ns), <code>truncQ=2</code>, <code>rm.phix=TRUE</code> and <code>maxEE=2</code>.
The maxEE parameter sets the maximum number of “expected errors” allowed in a read, which <a href="http://www.drive5.com/usearch/manual/expected_errors.html">according to the USEARCH authors</a> is a better filter than simply averaging quality scores.</p>
<div class="codehilite"><pre><span></span><span class="n">out</span> <span class="o">&lt;-</span> <span class="nf">filterAndTrim</span><span class="p">(</span><span class="n">raw_forward</span><span class="p">,</span> <span class="n">filtered_forward</span><span class="p">,</span> <span class="n">raw_reverse</span><span class="p">,</span>
                     <span class="n">filtered_reverse</span><span class="p">,</span> <span class="n">truncLen</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">240</span><span class="p">,</span><span class="m">160</span><span class="p">),</span> <span class="n">maxN</span><span class="o">=</span><span class="m">0</span><span class="p">,</span>
                     <span class="n">maxEE</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="m">2</span><span class="p">),</span> <span class="n">truncQ</span><span class="o">=</span><span class="m">2</span><span class="p">,</span> <span class="n">rm.phix</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> <span class="n">compress</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span>
                     <span class="n">multithread</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">out</span><span class="p">)</span>
</pre></div>


<h2 id="learn-the-error-rates">Learn the Error Rates<a class="headerlink" href="#learn-the-error-rates" title="Permanent link">&para;</a></h2>
<p>The DADA2 algorithm depends on a parametric error model and every amplicon dataset has a slightly different error rate.
The <code>learnErrors</code> of Dada2 learns the error model from the data and will help DADA2 to fits its method to your data</p>
<div class="codehilite"><pre><span></span><span class="n">errors_forward</span> <span class="o">&lt;-</span> <span class="nf">learnErrors</span><span class="p">(</span><span class="n">filtered_forward</span><span class="p">,</span> <span class="n">multithread</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="n">errors_reverse</span> <span class="o">&lt;-</span> <span class="nf">learnErrors</span><span class="p">(</span><span class="n">filtered_reverse</span><span class="p">,</span> <span class="n">multithread</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</pre></div>


<p>then we visualise the estimated error rates</p>
<div class="codehilite"><pre><span></span><span class="nf">plotErrors</span><span class="p">(</span><span class="n">errors_forward</span><span class="p">,</span> <span class="n">nominalQ</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">theme_minimal</span><span class="p">()</span>
</pre></div>


<div class="admonition question">
<p class="admonition-title">Question</p>
<p>Do you think the error model fits your data correctly?</p>
</div>
<h2 id="dereplication">Dereplication<a class="headerlink" href="#dereplication" title="Permanent link">&para;</a></h2>
<p>From the Dada2 documentation:</p>
<blockquote>
<p><strong>Dereplication</strong> combines all identical sequencing reads into into “unique sequences” with a corresponding “abundance”: the number of reads with that unique sequence.
<strong>Dereplication</strong> substantially reduces computation time by eliminating redundant comparisons.</p>
</blockquote>
<div class="codehilite"><pre><span></span><span class="n">derep_forward</span> <span class="o">&lt;-</span> <span class="nf">derepFastq</span><span class="p">(</span><span class="n">filtered_forward</span><span class="p">,</span> <span class="n">verbose</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="n">derep_reverse</span> <span class="o">&lt;-</span> <span class="nf">derepFastq</span><span class="p">(</span><span class="n">filtered_reverse</span><span class="p">,</span> <span class="n">verbose</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="c1"># name the derep-class objects by the sample names</span>
<span class="nf">names</span><span class="p">(</span><span class="n">derep_forward</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">sample_names</span>
<span class="nf">names</span><span class="p">(</span><span class="n">derep_reverse</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">sample_names</span>
</pre></div>


<h2 id="sample-inference">Sample inference<a class="headerlink" href="#sample-inference" title="Permanent link">&para;</a></h2>
<p>We are now ready to apply the core sequence-variant inference algorithm to the dereplicated data.</p>
<div class="codehilite"><pre><span></span><span class="n">dada_forward</span> <span class="o">&lt;-</span> <span class="nf">dada</span><span class="p">(</span><span class="n">derep_forward</span><span class="p">,</span> <span class="n">err</span><span class="o">=</span><span class="n">errors_forward</span><span class="p">,</span> <span class="n">multithread</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="n">dada_reverse</span> <span class="o">&lt;-</span> <span class="nf">dada</span><span class="p">(</span><span class="n">derep_reverse</span><span class="p">,</span> <span class="n">err</span><span class="o">=</span><span class="n">errors_reverse</span><span class="p">,</span> <span class="n">multithread</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>

<span class="c1"># inspect the dada-class object</span>
<span class="n">dada_forward[[1]]</span>
</pre></div>


<p>The DADA2 algorithm inferred 128 real sequence variants from the 1979 unique sequences in the first sample.</p>
<h2 id="merge-paired-end-reads">Merge Paired-end Reads<a class="headerlink" href="#merge-paired-end-reads" title="Permanent link">&para;</a></h2>
<p>Now that the reads are trimmed, dereplicated and error-corrected we can merge them together</p>
<div class="codehilite"><pre><span></span><span class="n">merged_reads</span> <span class="o">&lt;-</span> <span class="nf">mergePairs</span><span class="p">(</span><span class="n">dada_forward</span><span class="p">,</span> <span class="n">derep_forward</span><span class="p">,</span> <span class="n">dada_reverse</span><span class="p">,</span>
                           <span class="n">derep_reverse</span><span class="p">,</span> <span class="n">verbose</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>

<span class="c1"># inspect the merger data.frame from the first sample</span>
<span class="nf">head</span><span class="p">(</span><span class="n">merged_reads[[1]]</span><span class="p">)</span>
</pre></div>


<h2 id="construct-sequence-table">Construct Sequence Table<a class="headerlink" href="#construct-sequence-table" title="Permanent link">&para;</a></h2>
<p>We can now construct a sequence table of our mouse samples, a higher-resolution version of the OTU table produced by traditional methods.</p>
<div class="codehilite"><pre><span></span><span class="n">seq_table</span> <span class="o">&lt;-</span> <span class="nf">makeSequenceTable</span><span class="p">(</span><span class="n">merged_reads</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">seq_table</span><span class="p">)</span>

<span class="c1"># inspect distribution of sequence lengths</span>
<span class="nf">table</span><span class="p">(</span><span class="nf">nchar</span><span class="p">(</span><span class="nf">getSequences</span><span class="p">(</span><span class="n">seq_table</span><span class="p">)))</span>
</pre></div>


<h2 id="remove-chimeras">Remove Chimeras<a class="headerlink" href="#remove-chimeras" title="Permanent link">&para;</a></h2>
<p>The <code>dada</code> method used earlier removes substitutions and indel errors but chimeras remain.
We remove the chimeras with</p>
<div class="codehilite"><pre><span></span><span class="n">seq_table_nochim</span> <span class="o">&lt;-</span> <span class="nf">removeBimeraDenovo</span><span class="p">(</span><span class="n">seq_table</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s">&#39;consensus&#39;</span><span class="p">,</span>
                                       <span class="n">multithread</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> <span class="n">verbose</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">seq_table_nochim</span><span class="p">)</span>

<span class="c1"># which percentage of our reads did we keep?</span>
<span class="nf">sum</span><span class="p">(</span><span class="n">seq_table_nochim</span><span class="p">)</span> <span class="o">/</span> <span class="nf">sum</span><span class="p">(</span><span class="n">seq_table</span><span class="p">)</span>
</pre></div>


<p>As a final check of our progress, we’ll look at the number of reads that made it through each step in the pipeline</p>
<div class="codehilite"><pre><span></span><span class="n">get_n</span> <span class="o">&lt;-</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="nf">sum</span><span class="p">(</span><span class="nf">getUniques</span><span class="p">(</span><span class="n">x</span><span class="p">))</span>

<span class="n">track</span> <span class="o">&lt;-</span> <span class="nf">cbind</span><span class="p">(</span><span class="n">out</span><span class="p">,</span> <span class="nf">sapply</span><span class="p">(</span><span class="n">dada_forward</span><span class="p">,</span> <span class="n">get_n</span><span class="p">),</span> <span class="nf">sapply</span><span class="p">(</span><span class="n">merged_reads</span><span class="p">,</span> <span class="n">get_n</span><span class="p">),</span>
               <span class="nf">rowSums</span><span class="p">(</span><span class="n">seq_table</span><span class="p">),</span> <span class="nf">rowSums</span><span class="p">(</span><span class="n">seq_table_nochim</span><span class="p">))</span>

<span class="nf">colnames</span><span class="p">(</span><span class="n">track</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span> <span class="s">&#39;filtered&#39;</span><span class="p">,</span> <span class="s">&#39;denoised&#39;</span><span class="p">,</span> <span class="s">&#39;merged&#39;</span><span class="p">,</span> <span class="s">&#39;tabled&#39;</span><span class="p">,</span>
                     <span class="s">&#39;nonchim&#39;</span><span class="p">)</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">track</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">sample_names</span>
<span class="nf">head</span><span class="p">(</span><span class="n">track</span><span class="p">)</span>
</pre></div>


<p>We kept the majority of our reads!</p>
<h2 id="assign-taxonomy">Assign Taxonomy<a class="headerlink" href="#assign-taxonomy" title="Permanent link">&para;</a></h2>
<p>Now we assign taxonomy to our sequences using the SILVA database</p>
<div class="codehilite"><pre><span></span><span class="n">taxa</span> <span class="o">&lt;-</span> <span class="nf">assignTaxonomy</span><span class="p">(</span><span class="n">seq_table_nochim</span><span class="p">,</span>
                       <span class="s">&#39;MiSeq_SOP/silva_nr_v128_train_set.fa.gz&#39;</span><span class="p">,</span>
                       <span class="n">multithread</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="n">taxa</span> <span class="o">&lt;-</span> <span class="nf">addSpecies</span><span class="p">(</span><span class="n">taxa</span><span class="p">,</span> <span class="s">&#39;MiSeq_SOP/silva_species_assignment_v128.fa.gz&#39;</span><span class="p">)</span>
</pre></div>


<p>for inspecting the classification</p>
<div class="codehilite"><pre><span></span><span class="n">taxa_print</span> <span class="o">&lt;-</span> <span class="n">taxa</span>  <span class="c1"># removing sequence rownames for display only</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">taxa_print</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="kc">NULL</span>
<span class="nf">head</span><span class="p">(</span><span class="n">taxa_print</span><span class="p">)</span>
</pre></div>


<h2 id="phylogenetic-tree">Phylogenetic Tree<a class="headerlink" href="#phylogenetic-tree" title="Permanent link">&para;</a></h2>
<p>DADA2 is reference-free so we have to build the tree ourselves</p>
<p>We first align our sequences</p>
<div class="codehilite"><pre><span></span><span class="n">sequences</span> <span class="o">&lt;-</span> <span class="nf">getSequences</span><span class="p">(</span><span class="n">seq_table</span><span class="p">)</span>
<span class="nf">names</span><span class="p">(</span><span class="n">sequences</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">sequences</span>  <span class="c1"># this propagates to the tip labels of the tree</span>
<span class="n">alignment</span> <span class="o">&lt;-</span> <span class="nf">AlignSeqs</span><span class="p">(</span><span class="nf">DNAStringSet</span><span class="p">(</span><span class="n">sequences</span><span class="p">),</span> <span class="n">anchor</span><span class="o">=</span><span class="kc">NA</span><span class="p">)</span>
</pre></div>


<p>Then we build a neighbour-joining tree then fit a maximum likelihood tree using the neighbour-joining tree as a starting point</p>
<div class="codehilite"><pre><span></span><span class="n">phang_align</span> <span class="o">&lt;-</span> <span class="nf">phyDat</span><span class="p">(</span><span class="nf">as</span><span class="p">(</span><span class="n">alignment</span><span class="p">,</span> <span class="s">&#39;matrix&#39;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&#39;DNA&#39;</span><span class="p">)</span>
<span class="n">dm</span> <span class="o">&lt;-</span> <span class="nf">dist.ml</span><span class="p">(</span><span class="n">phang_align</span><span class="p">)</span>
<span class="n">treeNJ</span> <span class="o">&lt;-</span> <span class="nf">NJ</span><span class="p">(</span><span class="n">dm</span><span class="p">)</span>  <span class="c1"># note, tip order != sequence order</span>
<span class="n">fit</span> <span class="o">=</span> <span class="nf">pml</span><span class="p">(</span><span class="n">treeNJ</span><span class="p">,</span> <span class="n">data</span><span class="o">=</span><span class="n">phang_align</span><span class="p">)</span>

<span class="c1">## negative edges length changed to 0!</span>

<span class="n">fitGTR</span> <span class="o">&lt;-</span> <span class="nf">update</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span> <span class="n">k</span><span class="o">=</span><span class="m">4</span><span class="p">,</span> <span class="n">inv</span><span class="o">=</span><span class="m">0.2</span><span class="p">)</span>
<span class="n">fitGTR</span> <span class="o">&lt;-</span> <span class="nf">optim.pml</span><span class="p">(</span><span class="n">fitGTR</span><span class="p">,</span> <span class="n">model</span><span class="o">=</span><span class="s">&#39;GTR&#39;</span><span class="p">,</span> <span class="n">optInv</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> <span class="n">optGamma</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span>
                    <span class="n">rearrangement</span> <span class="o">=</span> <span class="s">&#39;stochastic&#39;</span><span class="p">,</span>
                    <span class="n">control</span> <span class="o">=</span> <span class="nf">pml.control</span><span class="p">(</span><span class="n">trace</span> <span class="o">=</span> <span class="m">0</span><span class="p">))</span>
<span class="nf">detach</span><span class="p">(</span><span class="s">&#39;package:phangorn&#39;</span><span class="p">,</span> <span class="n">unload</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</pre></div>


<h2 id="phyloseq">Phyloseq<a class="headerlink" href="#phyloseq" title="Permanent link">&para;</a></h2>
<p>First load the metadata</p>
<div class="codehilite"><pre><span></span><span class="n">sample_data</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span>
    <span class="s">&#39;https://hadrieng.github.io/tutorials/data/16S_metadata.txt&#39;</span><span class="p">,</span>
    <span class="n">header</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> <span class="n">row.names</span><span class="o">=</span><span class="s">&quot;sample_name&quot;</span><span class="p">)</span>
</pre></div>


<p>We can now construct a phyloseq object from our output and newly created metadata</p>
<div class="codehilite"><pre><span></span><span class="n">physeq</span> <span class="o">&lt;-</span> <span class="nf">phyloseq</span><span class="p">(</span><span class="nf">otu_table</span><span class="p">(</span><span class="n">seq_table_nochim</span><span class="p">,</span> <span class="n">taxa_are_rows</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">),</span>
                   <span class="nf">sample_data</span><span class="p">(</span><span class="n">sample_data</span><span class="p">),</span>
                   <span class="nf">tax_table</span><span class="p">(</span><span class="n">taxa</span><span class="p">),</span>
                   <span class="nf">phy_tree</span><span class="p">(</span><span class="n">fitGTR</span><span class="o">$</span><span class="n">tree</span><span class="p">))</span>
<span class="c1"># remove mock sample</span>
<span class="n">physeq</span> <span class="o">&lt;-</span> <span class="nf">prune_samples</span><span class="p">(</span><span class="nf">sample_names</span><span class="p">(</span><span class="n">physeq</span><span class="p">)</span> <span class="o">!=</span> <span class="s">&#39;Mock&#39;</span><span class="p">,</span> <span class="n">physeq</span><span class="p">)</span>
<span class="n">physeq</span>
</pre></div>


<p>Let's look at the alpha diversity of our samples</p>
<div class="codehilite"><pre><span></span><span class="nf">plot_richness</span><span class="p">(</span><span class="n">physeq</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s">&#39;day&#39;</span><span class="p">,</span> <span class="n">measures</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;Shannon&#39;</span><span class="p">,</span> <span class="s">&#39;Fisher&#39;</span><span class="p">),</span> <span class="n">color</span><span class="o">=</span><span class="s">&#39;when&#39;</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">theme_minimal</span><span class="p">()</span>
</pre></div>


<p>No obvious differences. Let's look at ordination methods (beta diversity)</p>
<p>We can perform an MDS with euclidean distance (mathematically equivalent to a PCA)</p>
<div class="codehilite"><pre><span></span><span class="n">ord</span> <span class="o">&lt;-</span> <span class="nf">ordinate</span><span class="p">(</span><span class="n">physeq</span><span class="p">,</span> <span class="s">&#39;MDS&#39;</span><span class="p">,</span> <span class="s">&#39;euclidean&#39;</span><span class="p">)</span>
<span class="nf">plot_ordination</span><span class="p">(</span><span class="n">physeq</span><span class="p">,</span> <span class="n">ord</span><span class="p">,</span> <span class="n">type</span><span class="o">=</span><span class="s">&#39;samples&#39;</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">&#39;when&#39;</span><span class="p">,</span>
                <span class="n">title</span><span class="o">=</span><span class="s">&#39;PCA of the samples from the MiSeq SOP&#39;</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">theme_minimal</span><span class="p">()</span>
</pre></div>


<p>now with the Bray-Curtis distance</p>
<div class="codehilite"><pre><span></span><span class="n">ord</span> <span class="o">&lt;-</span> <span class="nf">ordinate</span><span class="p">(</span><span class="n">physeq</span><span class="p">,</span> <span class="s">&#39;NMDS&#39;</span><span class="p">,</span> <span class="s">&#39;bray&#39;</span><span class="p">)</span>
<span class="nf">plot_ordination</span><span class="p">(</span><span class="n">physeq</span><span class="p">,</span> <span class="n">ord</span><span class="p">,</span> <span class="n">type</span><span class="o">=</span><span class="s">&#39;samples&#39;</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">&#39;when&#39;</span><span class="p">,</span>
                <span class="n">title</span><span class="o">=</span><span class="s">&#39;PCA of the samples from the MiSeq SOP&#39;</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">theme_minimal</span><span class="p">()</span>
</pre></div>


<p>There we can see a clear difference between our samples.</p>
<p>Let us take a look a the distribution of the most abundant families</p>
<div class="codehilite"><pre><span></span><span class="n">top20</span> <span class="o">&lt;-</span> <span class="nf">names</span><span class="p">(</span><span class="nf">sort</span><span class="p">(</span><span class="nf">taxa_sums</span><span class="p">(</span><span class="n">physeq</span><span class="p">),</span> <span class="n">decreasing</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">))</span><span class="n">[1</span><span class="o">:</span><span class="m">20</span><span class="n">]</span>
<span class="n">physeq_top20</span> <span class="o">&lt;-</span> <span class="nf">transform_sample_counts</span><span class="p">(</span><span class="n">physeq</span><span class="p">,</span> <span class="nf">function</span><span class="p">(</span><span class="n">OTU</span><span class="p">)</span> <span class="n">OTU</span><span class="o">/</span><span class="nf">sum</span><span class="p">(</span><span class="n">OTU</span><span class="p">))</span>
<span class="n">physeq_top20</span> <span class="o">&lt;-</span> <span class="nf">prune_taxa</span><span class="p">(</span><span class="n">top20</span><span class="p">,</span> <span class="n">physeq_top20</span><span class="p">)</span>
<span class="nf">plot_bar</span><span class="p">(</span><span class="n">physeq_top20</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s">&#39;day&#39;</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="s">&#39;Family&#39;</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">facet_wrap</span><span class="p">(</span><span class="o">~</span><span class="n">when</span><span class="p">,</span> <span class="n">scales</span><span class="o">=</span><span class="s">&#39;free_x&#39;</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">theme_minimal</span><span class="p">()</span>
</pre></div>


<p>We can place them in a tree</p>
<div class="codehilite"><pre><span></span><span class="n">bacteroidetes</span> <span class="o">&lt;-</span> <span class="nf">subset_taxa</span><span class="p">(</span><span class="n">physeq</span><span class="p">,</span> <span class="n">Phylum</span> <span class="o">%in%</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;Bacteroidetes&#39;</span><span class="p">))</span>
<span class="nf">plot_tree</span><span class="p">(</span><span class="n">bacteroidetes</span><span class="p">,</span> <span class="n">ladderize</span><span class="o">=</span><span class="s">&#39;left&#39;</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="s">&#39;abundance&#39;</span><span class="p">,</span>
          <span class="n">color</span><span class="o">=</span><span class="s">&#39;when&#39;</span><span class="p">,</span> <span class="n">label.tips</span><span class="o">=</span><span class="s">&#39;Family&#39;</span><span class="p">)</span>
</pre></div>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../pan_genome/" title="Pan-Genome Analysis" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Pan-Genome Analysis
              </span>
            </div>
          </a>
        
        
          <a href="../wms/" title="Whole Metagenome Sequencing" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Whole Metagenome Sequencing
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
  <div class="md-footer-social">
    <link rel="stylesheet" href="../assets/fonts/font-awesome.css">
    
      <a href="https://github.com/HadrienG" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://twitter.com/HGourle" class="md-footer-social__link fa fa-twitter"></a>
    
  </div>

    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.245445c6.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
    
  </body>
</html>