



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.4.0">
    
    
      
        <title>Pan-Genome Analysis - Bioinformatics Tutorials</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.0284f74d.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.01803549.css">
      
      
        
        
        <meta name="theme-color" content="#e91e63">
      
    
    
      <script src="../assets/javascripts/modernizr.74668098.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
    
    
      
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="pink" data-md-color-accent="pink">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#pan-genome-analysis" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href=".." title="Bioinformatics Tutorials" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Bioinformatics Tutorials
            </span>
            <span class="md-header-nav__topic">
              
                Pan-Genome Analysis
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/HadrienG/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    HadrienG/tutorials
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href=".." title="Bioinformatics Tutorials" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Bioinformatics Tutorials
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/HadrienG/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    HadrienG/tutorials
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Individual lessons
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Individual lessons
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../command_line/" title="The command-line" class="md-nav__link">
      The command-line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../file_formats/" title="File Formats" class="md-nav__link">
      File Formats
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../qc/" title="Quality Control and Trimming" class="md-nav__link">
      Quality Control and Trimming
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../mapping/" title="Mapping and Variant Calling" class="md-nav__link">
      Mapping and Variant Calling
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../assembly/" title="De-novo Genome Assembly" class="md-nav__link">
      De-novo Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../annotation/" title="Genome Annotation" class="md-nav__link">
      Genome Annotation
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Pan-Genome Analysis
      </label>
    
    <a href="./" title="Pan-Genome Analysis" class="md-nav__link md-nav__link--active">
      Pan-Genome Analysis
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#getting-the-data" title="Getting the data" class="md-nav__link">
    Getting the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assemble-and-annotate-the-strains" title="Assemble and Annotate the strains" class="md-nav__link">
    Assemble and Annotate the strains
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pan-genome-analysis_1" title="Pan-genome analysis" class="md-nav__link">
    Pan-genome analysis
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#plotting-the-result" title="Plotting the result" class="md-nav__link">
    Plotting the result
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../16S/" title="Metabarcoding" class="md-nav__link">
      Metabarcoding
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../wms/" title="Whole Metagenome Sequencing" class="md-nav__link">
      Whole Metagenome Sequencing
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../meta_assembly/" title="Metagenome assembly" class="md-nav__link">
      Metagenome assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../meta_assembly_2/" title="Metagenome assembly (continued)" class="md-nav__link">
      Metagenome assembly (continued)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../metavir/" title="Viral metagenomics" class="md-nav__link">
      Viral metagenomics
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../rna/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../nanopore/" title="Introduction to Nanopore Sequencing" class="md-nav__link">
      Introduction to Nanopore Sequencing
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#getting-the-data" title="Getting the data" class="md-nav__link">
    Getting the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assemble-and-annotate-the-strains" title="Assemble and Annotate the strains" class="md-nav__link">
    Assemble and Annotate the strains
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pan-genome-analysis_1" title="Pan-genome analysis" class="md-nav__link">
    Pan-genome analysis
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#plotting-the-result" title="Plotting the result" class="md-nav__link">
    Plotting the result
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/HadrienG/tutorials/edit/master/docs/pan_genome.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="pan-genome-analysis">Pan-Genome Analysis<a class="headerlink" href="#pan-genome-analysis" title="Permanent link">&para;</a></h1>
<p>In this tutorial we will learn how to determine a pan-genome from a collection of isolate genomes.</p>
<p>This tutorial is inspired from <a href="https://github.com/microgenomics/tutorials/blob/master/pangenome.md">Genome annotation and Pangenome Analysis</a> from the CBIB in Santiago, Chile</p>
<h2 id="getting-the-data">Getting the data<a class="headerlink" href="#getting-the-data" title="Permanent link">&para;</a></h2>
<p>We'll get data from <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1006258">this article</a> and analyse the core and accessory genomes of <em>E. coli</em>.</p>
<p>You can download raw data (paired-end reads) used to assemble a selection of the above-mentioned strains:</p>
<div class="codehilite"><pre><span></span>curl -O -J -L https://osf.io/s43mv/download
cat strains.txt <span class="p">|</span> parallel enaGroupGet -f fastq <span class="o">{}</span>
</pre></div>


<p>and then put all the reads in the same directory</p>
<div class="codehilite"><pre><span></span>mkdir reads
mv ERS*/*/*.fastq.gz reads/
rm -r ERS*
</pre></div>


<h2 id="assemble-and-annotate-the-strains">Assemble and Annotate the strains<a class="headerlink" href="#assemble-and-annotate-the-strains" title="Permanent link">&para;</a></h2>
<p>You'll assemble your strains with megahit.</p>
<div class="codehilite"><pre><span></span>mkdir assemblies
<span class="k">for</span> r1 in reads/*_1.fastq.gz
<span class="k">do</span>
    <span class="nv">prefix</span><span class="o">=</span><span class="k">$(</span>basename <span class="nv">$r1</span> _1.fastq.gz<span class="k">)</span>
    <span class="nv">r2</span><span class="o">=</span>reads/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_2.fastq.gz
    megahit -1 <span class="nv">$r1</span> -2 <span class="nv">$r2</span> -o <span class="si">${</span><span class="nv">prefix</span><span class="si">}</span> --out-prefix <span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>
    mv <span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.contigs.fa assemblies/
    rm -r <span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>
<span class="k">done</span>
</pre></div>


<p>and use prokka to annotate</p>
<div class="codehilite"><pre><span></span>mkdir annotation
<span class="k">for</span> assembly in assemblies/*.fa
<span class="k">do</span>
    <span class="nv">prefix</span><span class="o">=</span><span class="k">$(</span>basename <span class="nv">$assembly</span> .contigs.fa<span class="k">)</span>
    prokka --usegenus --genus Escherichia --species coli --strain <span class="si">${</span><span class="nv">prefix</span><span class="si">}</span> <span class="se">\</span>
        --outdir <span class="si">${</span><span class="nv">prefix</span><span class="si">}</span> --prefix <span class="si">${</span><span class="nv">prefix</span><span class="si">}</span> <span class="si">${</span><span class="nv">assembly</span><span class="si">}</span>
    mv <span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.gff annotation/
    rm -r <span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>
<span class="k">done</span>
</pre></div>


<h2 id="pan-genome-analysis_1">Pan-genome analysis<a class="headerlink" href="#pan-genome-analysis_1" title="Permanent link">&para;</a></h2>
<p>put all the .gff files in the same folder (e.g., <code>./gff</code>) and run Roary</p>
<div class="codehilite"><pre><span></span>roary -f roary -e -n -v annotation/*.gff
</pre></div>


<p>Roary will get all the coding sequences, convert them into protein, and create pre-clusters. Then, using BLASTP and MCL, Roary will create clusters, and check for paralogs. Finally, Roary will take every isolate and order them by presence/absence of orthologs. The summary output is present in the <code>summary_statistics.txt</code> file.</p>
<p>Additionally, Roary produces a <code>gene_presence_absence.csv</code> file that can be opened in any spreadsheet software to manually explore the results. In this file, you will find information such as gene name and gene annotation, and, of course, whether a gene is present in a genome or not.</p>
<h2 id="plotting-the-result">Plotting the result<a class="headerlink" href="#plotting-the-result" title="Permanent link">&para;</a></h2>
<p>Roary comes with a python script that allows you to generate a few plots to graphically assess your analysis output.</p>
<p>First, we need to generate a tree file from the alignment generated by Roary:</p>
<div class="codehilite"><pre><span></span><span class="n">cd</span> <span class="n">roary</span>
<span class="n">FastTree</span> <span class="o">-</span><span class="n">nt</span> <span class="o">-</span><span class="n">gtr</span> <span class="n">core_gene_alignment</span><span class="p">.</span><span class="n">aln</span> <span class="o">&gt;</span> <span class="n">my_tree</span><span class="p">.</span><span class="n">newick</span>
</pre></div>


<p>Then we can plot the Roary results with <code>roary_plots.py</code>, a community contributed python script to visualise roary results:</p>
<div class="codehilite"><pre><span></span><span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="p">.</span><span class="n">githubusercontent</span><span class="p">.</span><span class="n">com</span><span class="o">/</span><span class="n">sanger</span><span class="o">-</span><span class="n">pathogens</span><span class="o">/</span><span class="n">Roary</span><span class="o">/</span><span class="n">master</span><span class="o">/</span><span class="n">contrib</span><span class="o">/</span><span class="n">roary_plots</span><span class="o">/</span><span class="n">roary_plots</span><span class="p">.</span><span class="n">py</span>
<span class="n">python</span> <span class="n">roary_plots</span><span class="p">.</span><span class="n">py</span>
<span class="n">roary_plots</span><span class="p">.</span><span class="n">py</span> <span class="n">my_tree</span><span class="p">.</span><span class="n">newick</span> <span class="n">gene_presence_absence</span><span class="p">.</span><span class="n">csv</span>
</pre></div>


<p>then look at the 3 <code>/png</code> files that have been generated</p>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../annotation/" title="Genome Annotation" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Genome Annotation
              </span>
            </div>
          </a>
        
        
          <a href="../16S/" title="Metabarcoding" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Metabarcoding
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
  <div class="md-footer-social">
    <link rel="stylesheet" href="../assets/fonts/font-awesome.css">
    
      <a href="https://github.com/HadrienG" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://twitter.com/HGourle" class="md-footer-social__link fa fa-twitter"></a>
    
  </div>

    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.245445c6.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
    
  </body>
</html>