



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.4.0">
    
    
      
        <title>Quality Control and Trimming - Bioinformatics Tutorials</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.0284f74d.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.01803549.css">
      
      
        
        
        <meta name="theme-color" content="#e91e63">
      
    
    
      <script src="../assets/javascripts/modernizr.74668098.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
    
    
      
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="pink" data-md-color-accent="pink">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#quality-control-and-trimming" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href=".." title="Bioinformatics Tutorials" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Bioinformatics Tutorials
            </span>
            <span class="md-header-nav__topic">
              
                Quality Control and Trimming
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/HadrienG/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    HadrienG/tutorials
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href=".." title="Bioinformatics Tutorials" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Bioinformatics Tutorials
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/HadrienG/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    HadrienG/tutorials
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Individual lessons
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Individual lessons
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../command_line/" title="The command-line" class="md-nav__link">
      The command-line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../file_formats/" title="File Formats" class="md-nav__link">
      File Formats
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Quality Control and Trimming
      </label>
    
    <a href="./" title="Quality Control and Trimming" class="md-nav__link md-nav__link--active">
      Quality Control and Trimming
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#lecture" title="Lecture" class="md-nav__link">
    Lecture
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#practical" title="Practical" class="md-nav__link">
    Practical
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#downloading-the-data" title="Downloading the data" class="md-nav__link">
    Downloading the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#working-directory" title="Working Directory" class="md-nav__link">
    Working Directory
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fastqc" title="FastQC" class="md-nav__link">
    FastQC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#scythe" title="Scythe" class="md-nav__link">
    Scythe
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sickle" title="Sickle" class="md-nav__link">
    Sickle
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fastqc-again" title="FastQC again" class="md-nav__link">
    FastQC again
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#multiqc" title="MultiQC" class="md-nav__link">
    MultiQC
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../mapping/" title="Mapping and Variant Calling" class="md-nav__link">
      Mapping and Variant Calling
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../assembly/" title="De-novo Genome Assembly" class="md-nav__link">
      De-novo Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../annotation/" title="Genome Annotation" class="md-nav__link">
      Genome Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pan_genome/" title="Pan-Genome Analysis" class="md-nav__link">
      Pan-Genome Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../16S/" title="Metabarcoding" class="md-nav__link">
      Metabarcoding
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../wms/" title="Whole Metagenome Sequencing" class="md-nav__link">
      Whole Metagenome Sequencing
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../meta_assembly/" title="Metagenome assembly" class="md-nav__link">
      Metagenome assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../meta_assembly_2/" title="Metagenome assembly (continued)" class="md-nav__link">
      Metagenome assembly (continued)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../metavir/" title="Viral metagenomics" class="md-nav__link">
      Viral metagenomics
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../rna/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../nanopore/" title="Introduction to Nanopore Sequencing" class="md-nav__link">
      Introduction to Nanopore Sequencing
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#lecture" title="Lecture" class="md-nav__link">
    Lecture
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#practical" title="Practical" class="md-nav__link">
    Practical
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#downloading-the-data" title="Downloading the data" class="md-nav__link">
    Downloading the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#working-directory" title="Working Directory" class="md-nav__link">
    Working Directory
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fastqc" title="FastQC" class="md-nav__link">
    FastQC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#scythe" title="Scythe" class="md-nav__link">
    Scythe
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sickle" title="Sickle" class="md-nav__link">
    Sickle
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fastqc-again" title="FastQC again" class="md-nav__link">
    FastQC again
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#multiqc" title="MultiQC" class="md-nav__link">
    MultiQC
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/HadrienG/tutorials/edit/master/docs/qc.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="quality-control-and-trimming">Quality Control and Trimming<a class="headerlink" href="#quality-control-and-trimming" title="Permanent link">&para;</a></h1>
<h2 id="lecture">Lecture<a class="headerlink" href="#lecture" title="Permanent link">&para;</a></h2>
<p><br></p>
<iframe src="https://docs.google.com/presentation/d/e/2PACX-1vR7ZN5Gc-3q5DR8CWQgyHhjUIc-6uKJnB48lDVMB7tWrt4gpZFipbJRVWfaqdxSw9n_WnMfXoFMZXds/embed?start=false&loop=false&delayms=3000" frameborder="0" width="480" height="389" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe>

<h2 id="practical">Practical<a class="headerlink" href="#practical" title="Permanent link">&para;</a></h2>
<p>In this practical you will learn to import, view and check the quality of raw high thoughput sequencing sequencing data.</p>
<p>The first dataset you will be working with is from an Illumina MiSeq dataset.
The sequenced organism is an enterohaemorrhagic E. coli (EHEC) of the serotype O157, a potentially fatal gastrointestinal pathogen.
The sequenced bacterium was part of an outbreak investigation in the St. Louis area, USA in 2011.
The sequencing was done as paired-end 2x150bp.</p>
<h2 id="downloading-the-data">Downloading the data<a class="headerlink" href="#downloading-the-data" title="Permanent link">&para;</a></h2>
<p>The raw data were deposited at the European Nucleotide Archive, under the accession number SRR957824.
You could go to the ENA <a href="http://www.ebi.ac.uk/ena">website</a> and search for the run with the accession SRR957824.</p>
<p>However these files contain about 3 million reads and are therefore quite big.
We are only gonna use a subset of the original dataset for this tutorial.</p>
<p>First create a <code>data/</code> directory in your home folder</p>
<div class="codehilite"><pre><span></span>mkdir ~/data
</pre></div>


<p>now let's download the subset</p>
<div class="codehilite"><pre><span></span><span class="nb">cd</span> ~/data
curl -O -J -L https://osf.io/shqpv/download
curl -O -J -L https://osf.io/9m3ch/download
</pre></div>


<p>Let’s make sure we downloaded all of our data using md5sum.</p>
<div class="codehilite"><pre><span></span>md5sum SRR957824_500K_R1.fastq.gz SRR957824_500K_R2.fastq.gz
</pre></div>


<p>you should see this</p>
<div class="codehilite"><pre><span></span><span class="mi">1</span><span class="n">e8cf249e3217a5a0bcc0d8a654585fb</span>  <span class="n">SRR957824_500K_R1</span><span class="p">.</span><span class="n">fastq</span><span class="p">.</span><span class="n">gz</span>
<span class="mi">70</span><span class="n">c726a31f05f856fe942d727613adb7</span>  <span class="n">SRR957824_500K_R2</span><span class="p">.</span><span class="n">fastq</span><span class="p">.</span><span class="n">gz</span>
</pre></div>


<p>and now look at the file names and their size</p>
<div class="codehilite"><pre><span></span>ls -l
</pre></div>


<div class="codehilite"><pre><span></span><span class="n">total</span> <span class="mi">97</span><span class="n">M</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="c1">--r-- 1 hadrien 48M Nov 19 18:44 SRR957824_500K_R1.fastq.gz</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="c1">--r-- 1 hadrien 50M Nov 19 18:53 SRR957824_500K_R2.fastq.gz</span>
</pre></div>


<p>There are 500 000 paired-end reads taken randomly from the original data</p>
<p>One last thing before we get to the quality control: those files are writeable.
By default, UNIX makes things writeable by the file owner.
This poses an issue with creating typos or errors in raw data.
We fix that before going further</p>
<div class="codehilite"><pre><span></span>chmod u-w *
</pre></div>


<h2 id="working-directory">Working Directory<a class="headerlink" href="#working-directory" title="Permanent link">&para;</a></h2>
<p>First we make a work directory: a directory where we can play around with a copy of the data without messing with the original</p>
<div class="codehilite"><pre><span></span>mkdir ~/work
<span class="nb">cd</span> ~/work
</pre></div>


<p>Now we make a link of the data in our working directory</p>
<div class="codehilite"><pre><span></span>ln -s ~/data/* .
</pre></div>


<p>The files that we've downloaded are FASTQ files. Take a look at one of them with</p>
<div class="codehilite"><pre><span></span>zless SRR957824_500K_R1.fastq.gz
</pre></div>


<div class="admonition tip">
<p class="admonition-title">Tip</p>
</div>
<p>Use the spacebar to scroll down, and type ‘q’ to exit ‘less’</p>
<p>You can read more on the FASTQ format in the <a href="../file_formats/">File Formats</a> lesson.</p>
<div class="admonition question">
<p class="admonition-title">Question</p>
</div>
<p>Where does the filename come from?</p>
<div class="admonition question">
<p class="admonition-title">Question</p>
</div>
<p>Why are there 1 and 2 in the file names?</p>
<h2 id="fastqc">FastQC<a class="headerlink" href="#fastqc" title="Permanent link">&para;</a></h2>
<p>To check the quality of the sequence data we will use a tool called FastQC.</p>
<p>FastQC has a graphical interface and can be downloaded and run on a Windows or Linux computer without installation.
It is available <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">here</a>.</p>
<p>However, FastQC is also available as a command line utility on the training server you are using.
To run FastQC on our two files</p>
<div class="codehilite"><pre><span></span>fastqc SRR957824_500K_R1.fastq.gz SRR957824_500K_R2.fastq.gz
</pre></div>


<p>and look what FastQC has produced</p>
<div class="codehilite"><pre><span></span><span class="n">ls</span> <span class="o">*</span><span class="n">fastqc</span><span class="o">*</span>
</pre></div>


<p>For each file, FastQC has produced both a .zip archive containing all the plots, and a html report.</p>
<p>Download and open the html files with your favourite web browser.</p>
<p>Alternatively you can look a these copies of them:</p>
<ul>
<li><a href="../data/fastqc/SRR957824_500K_R1_fastqc.html">SRR957824_500K_R1_fastqc.html</a></li>
<li><a href="../data/fastqc/SRR957824_500K_R2_fastqc.html">SRR957824_500K_R2_fastqc.html</a></li>
</ul>
<div class="admonition question">
<p class="admonition-title">Question</p>
</div>
<p>What should you pay attention to in the FastQC report?</p>
<div class="admonition question">
<p class="admonition-title">Question</p>
</div>
<p>Which file is of better quality?</p>
<p>Pay special attention to the per base sequence quality and sequence length distribution.
Explanations for the various quality modules can be found <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/">here</a>.
Also, have a look at examples of a <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html">good</a> and a <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html">bad</a> illumina read set for comparison.</p>
<p>You will note that the reads in your uploaded dataset have fairly poor quality (&lt;20) towards the end. There are also outlier reads that have very poor quality for most of the second half of the reads.</p>
<h2 id="scythe">Scythe<a class="headerlink" href="#scythe" title="Permanent link">&para;</a></h2>
<p>Now we'll do some trimming!</p>
<p>Scythe uses a Naive Bayesian approach to classify contaminant substrings in sequence reads.
It considers quality information, which can make it robust in picking out 3'-end adapters, which often include poor quality bases.</p>
<p>The first thing we need is the adapters to trim off</p>
<div class="codehilite"><pre><span></span>curl -O -J -L https://osf.io/v24pt/download
</pre></div>


<p>Now we run scythe on both our read files</p>
<div class="codehilite"><pre><span></span>scythe -a adapters.fasta -o SRR957824_adapt_R1.fastq SRR957824_500K_R1.fastq.gz
scythe -a adapters.fasta -o SRR957824_adapt_R2.fastq SRR957824_500K_R2.fastq.gz
</pre></div>


<div class="admonition question">
<p class="admonition-title">Question</p>
</div>
<p>What adapters do you use?</p>
<h2 id="sickle">Sickle<a class="headerlink" href="#sickle" title="Permanent link">&para;</a></h2>
<p>Most modern sequencing technologies produce reads that have deteriorating quality towards the 3'-end and some towards the 5'-end as well.
Incorrectly called bases in both regions negatively impact assembles, mapping, and downstream bioinformatics analyses.</p>
<p>We will trim each read individually down to the good quality part to keep the bad part from interfering with downstream applications.</p>
<p>To do so, we will use sickle. Sickle is a tool that uses sliding windows along with quality and length thresholds to determine when quality is sufficiently low to trim the 3'-end of reads and also determines when the quality is sufficiently high enough to trim the 5'-end of reads. It will also discard reads based upon a length threshold.</p>
<p>To run sickle</p>
<div class="codehilite"><pre><span></span>sickle pe -f SRR957824_adapt_R1.fastq -r SRR957824_adapt_R2.fastq <span class="se">\</span>
    -t sanger -o SRR957824_trimmed_R1.fastq -p SRR957824_trimmed_R2.fastq <span class="se">\</span>
    -s /dev/null -q <span class="m">25</span>
</pre></div>


<p>which should output something like</p>
<div class="codehilite"><pre><span></span><span class="n">PE</span> <span class="k">forward</span> <span class="n">file</span><span class="p">:</span> <span class="n">SRR957824_trimmed_R1</span><span class="p">.</span><span class="n">fastq</span>
<span class="n">PE</span> <span class="n">reverse</span> <span class="n">file</span><span class="p">:</span> <span class="n">SRR957824_trimmed_R2</span><span class="p">.</span><span class="n">fastq</span>

<span class="n">Total</span> <span class="k">input</span> <span class="n">FastQ</span> <span class="n">records</span><span class="p">:</span> <span class="mi">1000000</span> <span class="p">(</span><span class="mi">500000</span> <span class="n">pairs</span><span class="p">)</span>

<span class="n">FastQ</span> <span class="n">paired</span> <span class="n">records</span> <span class="n">kept</span><span class="p">:</span> <span class="mi">834570</span> <span class="p">(</span><span class="mi">417285</span> <span class="n">pairs</span><span class="p">)</span>
<span class="n">FastQ</span> <span class="n">single</span> <span class="n">records</span> <span class="n">kept</span><span class="p">:</span> <span class="mi">13263</span> <span class="p">(</span><span class="k">from</span> <span class="n">PE1</span><span class="p">:</span> <span class="mi">11094</span><span class="p">,</span> <span class="k">from</span> <span class="n">PE2</span><span class="p">:</span> <span class="mi">2169</span><span class="p">)</span>
<span class="n">FastQ</span> <span class="n">paired</span> <span class="n">records</span> <span class="n">discarded</span><span class="p">:</span> <span class="mi">138904</span> <span class="p">(</span><span class="mi">69452</span> <span class="n">pairs</span><span class="p">)</span>
<span class="n">FastQ</span> <span class="n">single</span> <span class="n">records</span> <span class="n">discarded</span><span class="p">:</span> <span class="mi">13263</span> <span class="p">(</span><span class="k">from</span> <span class="n">PE1</span><span class="p">:</span> <span class="mi">2169</span><span class="p">,</span> <span class="k">from</span> <span class="n">PE2</span><span class="p">:</span> <span class="mi">11094</span><span class="p">)</span>
</pre></div>


<h2 id="fastqc-again">FastQC again<a class="headerlink" href="#fastqc-again" title="Permanent link">&para;</a></h2>
<p>Run fastqc again on the filtered reads</p>
<div class="codehilite"><pre><span></span>fastqc SRR957824_trimmed_R1.fastq SRR957824_trimmed_R2.fastq
</pre></div>


<p>and look at the reports</p>
<ul>
<li><a href="../data/fastqc/SRR957824_trimmed_R1_fastqc.html">SRR957824_trimmed_R1_fastqc.html</a></li>
<li><a href="../data/fastqc/SRR957824_trimmed_R2_fastqc.html">SRR957824_trimmed_R2_fastqc.html</a></li>
</ul>
<h2 id="multiqc">MultiQC<a class="headerlink" href="#multiqc" title="Permanent link">&para;</a></h2>
<p><a href="http://multiqc.info">MultiQC</a> is a tool that aggreagtes results from several popular QC bioinformatics software into one html report.</p>
<p>Let's run MultiQC in our current directory</p>
<div class="codehilite"><pre><span></span>multiqc .
</pre></div>


<p>You can download the report or view it by clickinh on the link below</p>
<ul>
<li><a href="../data/fastqc/multiqc_report.html">multiqc_report.html</a></li>
</ul>
<div class="admonition question">
<p class="admonition-title">Question</p>
</div>
<p>What did the trimming do to the per-base sequence quality, the per sequence quality scores and the sequence length distribution?</p>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../file_formats/" title="File Formats" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                File Formats
              </span>
            </div>
          </a>
        
        
          <a href="../mapping/" title="Mapping and Variant Calling" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Mapping and Variant Calling
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
  <div class="md-footer-social">
    <link rel="stylesheet" href="../assets/fonts/font-awesome.css">
    
      <a href="https://github.com/HadrienG" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://twitter.com/HGourle" class="md-footer-social__link fa fa-twitter"></a>
    
  </div>

    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.245445c6.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
    
  </body>
</html>